src/com/google/javascript/jscomp/PolymerProtectStaticProperties.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+
+/**
+ * Prevents renaming of the Polymer 2 static properties "is" and "config" by adding externs to the
+ * Function.prototype.
+ *
+ * <p>This pass must run after type checking since otherwise all functions would have an "is" and
+ * "config" property.
+ */
+class PolymerProtectStaticProperties implements CompilerPass {
+
+  private final AbstractCompiler compiler;
+
+  PolymerProtectStaticProperties(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+    FindPolymerExterns findPolymerExterns = new FindPolymerExterns();
+    NodeTraversal.traverseEs6(compiler, externs, findPolymerExterns);
+
+    if (findPolymerExterns.hasPolymerElementExterns()) {
+      addExtern("is");
```

Comment:
We may add optional mixins that support `hostAttributes` and `listener` static properties. How should we handle those? The set of important statics is nearly as open as the set of instance methods now. Is there any way the _users_ of these properties can protect them or allow for renaming?

That seems to be what this is attempting: https://github.com/Polymer/polymer/blob/master/lib/mixins/element-mixin.html#L141

Is this not sufficient?